### PR TITLE
self-run-agents: install dotnet runtime

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: self-run-agents/ami-build/agent-setup.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}' | sudo tee /etc/docker/daemon.json
+
+sudo systemctl enable docker
+sudo systemctl start docker
+
+sudo useradd -ms /bin/bash -G docker azure-pipelines
+sudo mkdir -p /srv/azure-pipelines
+sudo chown -R azure-pipelines:azure-pipelines /srv/azure-pipelines/
+
+if [[ "${ARCH}" == "amd64" ]]; then
```
Comment (SecurityInsanity) [inline self-run-agents/ami-build/agent-setup.sh:33]: Does ARM, not need dotnet runtime? Why does x86 need it?

File: self-run-agents/ami-build/agent-setup.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+wget -q -O - https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo apt-key adv --list-public-keys --with-fingerprint --with-colons 0EBFCD88 2>/dev/null | grep 'fpr' | head -n1 | grep '9DC858229FC7DD38854AE2D88D81803C0EBFCD88'
+sudo add-apt-repository -y "deb [arch=${ARCH}] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+sudo apt-add-repository -y ppa:git-core/ppa
+
+sudo apt-get update
+sudo apt-get install -y docker-ce docker-ce-cli containerd.io libunwind8 libcurl3 git awscli jq
+
+sudo mkdir -p /etc/docker
+echo '{
```
Comment (SecurityInsanity) [inline self-run-agents/ami-build/agent-setup.sh:21]: What's this for?

File: self-run-agents/ami-build/agent-setup.sh (hunk 1 above)
Comment (lizan) [reply self-run-agents/ami-build/agent-setup.sh:33]: MS doesn't release dotnet runtime in apt. This is only for coverage result publish: https://dev.azure.com/cncf/envoy/_build/results?buildId=39829&view=logs&j=1117f030-3e3a-5822-cfd7-97e776c1f4cb&t=e7a432d4-ba87-5541-286f-122716d8d1ce

File: self-run-agents/ami-build/agent-setup.sh (hunk 2 above)
Comment (lizan) [reply self-run-agents/ami-build/agent-setup.sh:21]: https://github.com/envoyproxy/envoy/blob/master/.azure-pipelines/bazel.yml#L35-L43

**Pull request merged**